src/elf_loader.hpp

```
@@ -21,20 +22,59 @@ int create_map_crab(const EbpfMapType& map_type, uint32_t key_size, uint32_t val
-                                 const std::string& desired_program, const ebpf_verifier_options_t& options,
-                                 const ebpf_platform_t* platform);
+struct ElfProgramInfo {
+    std::string section_name;
+    std::string function_name;
+    uint32_t section_offset{};
+    bool invalid{};
+    std::string invalid_reason;
+};
+
+class ElfObject {
+    struct QueryKey {
```

Clang (14.0.6)

[warning] constructor does not initialize these fields: section, program

(cppcoreguidelines-pro-type-member-init)